src/lib/extension-load-queue.ts

```diff
@@ -1,63 +1,149 @@
+/**
+ * This file provides utilities for managing and processing load queues in a browser environment.
+ *
+ * It includes:
+ * - A system for creating and managing load queues identified by unique symbols.
+ * - A script store that holds callbacks associated with controllers.
+ * - A React hook for using and managing controllers within a script store.
+ *
+ * Example usage:
+ *
+ * const MY_QUEUE_SYMBOL = Symbol.for('my-queue');
+ * const MY_STORE_SYMBOL = Symbol.for('my-controllers-store');
+ *
+ * const myStore = getScriptStore(MY_STORE_SYMBOL);
+ *
+ * createLoadQueue({
+ *   store: myStore,
+ *   createController: () => new MyController(),
+ *   queueKey: MY_QUEUE_SYMBOL,
+ * });
+ *
+ * const controller = useController(myStore);
+ */
+
 import {useEffect, useState} from 'react';
 
+import {isBrowser} from './browser';
+
 export type ControllerLoadedCallback<T> = (controller: T) => void;
 
-export const QUEUE_SYMBOL = Symbol.for('queue');
+export const QUEUES_SYMBOL = Symbol.for('extension-load-queues');
+export const SINGLE_QUEUE_SYMBOL = Symbol.for('single-queue');
 
 export type ScriptStore<T> = ControllerLoadedCallback<T>[];
 
+/**
+ * Interface for creating a load queue.
+ */
 export interface CreateLoadQueueArgs<T> {
+    // The store where callbacks are saved
     store: ScriptStore<T>;
+
+    // Function that creates a controller
     createController: () => T;
+
+    // Flag to check if the queue is already created
     isQueueCreated?: boolean;
+
+    // Callback to handle queue creation
     onQueueCreated?: (created: boolean) => void;
-}
 
-// TODO: this function is a very frequently used utility,
-//  so we should place it in a separate top-level file and document it.
-export const isBrowser = () => typeof window !== 'undefined' && typeof document !== 'undefined';
+    // Symbol key for identifying the queue
+    queueKey?: symbol;
+}
 
-// TODO: window casts are weird — fix encounter any issues with declaration merging
-export const getScriptStore = <T>(key: symbol): ScriptStore<T> => {
+/**
+ * Retrieves or initializes a script store for the given symbol key.
+ *
+ * @param {symbol} storeKey - The key associated with the script store.
+ * @returns {ScriptStore<T>} The script store associated with the given key.
+ */
+export const getScriptStore = <T>(storeKey: symbol): ScriptStore<T> => {
     if (isBrowser()) {
-        (window as Window)[key] = (window as Window)[key] || [];
-        return (window as Window)[key];
+        // Initialize the store if it does not exist or check that it is an array
```

Comment:
In my opinion, this commenting style is too verbose. JSDoc is fine, but I don't think we really need to comment on every other statement.